cdci_data_analysis/analysis/instrument.py

```diff
@@ -155,45 +156,76 @@ def parse_inputs_files(self,
                              verbose,
                              use_scws,
                              sentry_client=None):
+        # if any, for exceptions additional imformation
+        content_temp_dir = None
         # TODO probably exception handling can be further improved and/or optmized
         # set catalog
         try:
             self.upload_catalog_from_fronted(par_dic=par_dic, request=request, temp_dir=temp_dir)
         except Exception as e:
+            if temp_dir is not None and os.path.exists(temp_dir):
+                content_temp_dir = os.listdir(temp_dir)
+            error_message = 'Error while uploading catalog file from the frontend'
+            if content_temp_dir is not None:
+                error_message += f', content of the temporary directory is {content_temp_dir}'
```

Comment:
this block is almost exactly repeated 5 times. It can be made in context or a decorator. Or even just single try-except block (in this case it to keep the same message we'd need to store the couple of changing words in a variable)

Reply:
Removed now quite some code repetition.

Reply:
But of course I'm always open to more improvements